data-structures/stacks-1.ts

```diff
@@ -0,0 +1,28 @@
+/* eslint-disable @typescript-eslint/no-empty-function */
+import { Stack } from './lib/stack';
+
+export function getTop<T>(stack: Stack<T>): T | undefined {
+  const result = stack.peek();
+  if (!stack) {
```

Comment:
I don't think this is what you intended, since according to the TypeScript type, `stack` can never be falsy. Also, if it were falsy, in the line above `stack.peek()` would throw an exception.

I think you meant to write `if (!result)`. But that one isn't necessary, either, since `stack.peek()` will return undefined if the stack is empty, so you can just `return stack.peek();`.